pinot-core/src/main/java/org/apache/pinot/core/segment/index/datasource/BaseDataSource.java

```diff
@@ -46,19 +48,29 @@ public BaseDataSource(DataSourceMetadata dataSourceMetadata, ForwardIndexReader<
       @Nullable Dictionary dictionary, @Nullable InvertedIndexReader<?> invertedIndex,
       @Nullable InvertedIndexReader<?> rangeIndex, @Nullable TextIndexReader textIndex,
       @Nullable TextIndexReader fstIndex, @Nullable JsonIndexReader jsonIndex, @Nullable BloomFilterReader bloomFilter,
-      @Nullable NullValueVectorReader nullValueVector) {
+      @Nullable NullValueVectorReader nullValueVector, @Nullable H3IndexReader h3Index) {
```

Comment:
Put argument `h3Index` between `rangeIndex` and `textIndex`

Reply:
this is actually intentional to use append. It's easier for subclass to determine the differences.

Reply:
Since all the subclasses are internal, we'd better keep them the same order for easier maintenance in the future as we add more indexes.